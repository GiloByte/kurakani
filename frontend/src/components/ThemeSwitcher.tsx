"use client";

import { useState, useEffect } from "react";
import { useTheme } from "next-themes";
import { SunIcon } from '@heroicons/react/24/outline';
import { MoonIcon } from '@heroicons/react/24/solid';


const ThemeSwitcher = () => {
  const [mounted, setMounted] = useState(false);
  const {systemTheme, theme, setTheme } = useTheme();

  // useEffect only runs on the client, so now we can safely show the UI
  useEffect(() => {
    setMounted(true);
  }, []);

  if (!mounted) {
    return null;
  }

  const renderThemeChanger= () => {
    if(!mounted) return null;

    const currentTheme = theme === "system" ? systemTheme : theme ;

    if(currentTheme ==="dark"){
      return (
        <SunIcon className="w-6 h-6 text-yellow-500 " role="button" onClick={() =>
         setTheme('light')} />
      )
    }

    else {
      return (
        <MoonIcon className="w-6 h-6 text-gray-900" role="button" onClick={() =>
         setTheme('dark')} />
      )
    }
 };

  return (
    <>
    {renderThemeChanger()}
    </>
  );
};

export default ThemeSwitcher;