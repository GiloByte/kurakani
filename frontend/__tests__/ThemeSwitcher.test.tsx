import React from 'react';
import '@testing-library/jest-dom'
import { render, screen, fireEvent } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import ThemeSwitcher from '../src/components/ThemeSwitcher';

test('initializes with mounted state set to true', () => {
  render(<ThemeSwitcher />);
  const sunIcon = screen.queryByTestId('sun-icon'); // Použijeme queryByTestId, aby sme nevyhodili chybu, ak neexistuje
  const moonIcon = screen.getByTestId('moon-icon');
  expect(sunIcon).toBeNull(); // Očakávame, že sun-icon nebude v dokumente
  expect(moonIcon).toBeInTheDocument();
});

// test('calls setTheme with "light" when dark mode icon is clicked', () => {
//   render(<ThemeSwitcher />);
//   const moonIcon = screen.queryByTestId('moon-icon');
//   fireEvent.click(moonIcon!);
// });

test('renders ThemeSwitcher correctly', () => {
  const tree = TestRenderer.create(<ThemeSwitcher />).toJSON();
  expect(tree).toMatchSnapshot();
});




import { act } from 'react-dom/test-utils';

test('calls setTheme with "light" when dark mode icon is clicked', () => {
  // Získať referenciu na funkciu setTheme zo závislosti (ak je dostupná)
  const setThemeMock = jest.fn();
  jest.mock('../src/components/ThemeSwitcher.tsx', () => ({
    useTheme: () => ({ setTheme: setThemeMock }),
  }));

  // Renderovať komponent s novým kontextom
  render(<ThemeSwitcher />);

  // Simulovať kliknutie na ikonu tmavého režimu
  const moonIcon = screen.queryByTestId('moon-icon');
  fireEvent.click(moonIcon!);

  // Simulovať nárast pamäťovej záťaže (príklad s použitím fake dát)
  act(() => {
    for (let i = 0; i < 10000; i++) {
      // Predstavme si, že tu dochádza k vytváraniu a manipulácii s dátami
      setThemeMock(`light-${i}`);
    }
  });

  // Overiť, či bolo volanie setTheme s argumentom, ktorý začína s "light", spustené
  expect(setThemeMock).toHaveBeenCalledWith(expect.stringMatching(/^light-/));
});
