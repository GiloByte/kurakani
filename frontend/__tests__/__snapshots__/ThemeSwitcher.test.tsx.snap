// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ThemeSwitcher correctly 1`] = `null`;
